azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -130,10 +220,33 @@ public long getLastCleanerThreadCheckTime() {
   }
 
   @Override
+  public Set<Executor> getAllActiveExecutors() {
+    return activeExecutors;
+  }
+
+  /**
+   *
+   * {@inheritDoc}
+   *
+   * @see azkaban.executor.ExecutorManagerAdapter#fetchExecutor(int)
+   */
+  @Override
+  public Executor fetchExecutor(int executorId) throws ExecutorManagerException {
+    for (Executor executor : activeExecutors) {
+      if (executor.getId() == executorId) {
+        return executor;
+      }
+    }
+    return executorLoader.fetchExecutor(executorId);
+  }
+
+  @Override
   public Set<String> getPrimaryServerHosts() {
-    // Only one for now. More probably later.
+    // TODO: do we want to have a primary
     HashSet<String> ports = new HashSet<String>();
```

Comment:
is variable name "ports" a misnomer? As we seem to be putting server:port pair into the HashSet